contracts/interchain-token-service/src/contract.rs

```diff
@@ -509,11 +493,11 @@ impl InterchainTokenService {
                 token_handler::give_token(
                     env,
                     &destination_address,
-                    Self::token_id_config(env, token_id.clone()),
+                    Self::token_id_config(env, token_id.clone())?,
                     amount,
                 )?;
 
-                let token_address = Self::token_address(env, token_id.clone());
+                let token_address = Self::token_address(env, token_id.clone())?;
```

Comment:
we're querying the token id twice. query it once, and retrieve the token address from within it

Reply:
yeah.. that's true. updated.